tests/test_dse.py

```diff
@@ -20,6 +21,42 @@
 pytestmark = skipif(['yask', 'ops'])
 
 
+@pytest.mark.parametrize('expected', [
```

Comment:
instead of writing out the entire Eqs, what if we had just the LHSs? 

```
@pytest.mark.parametrize('expected', [(r3, r2, d(t,x,y), a(...), ...)])
```

this would make it much simpler to add new cases

I think also this test is nice, but we could add some easier cases before this one. Indeed, the four equations [eq1, eq4] should be part of the parametrization as well

Reply:
I prefer it like that, it's a bit hard to figure out what's going on with just he LHS names.

Reply:
well this makes the test hard to extend... and AFAICT the RHSs do not add any information ?

Reply:
It makes it readable, can't understand/see why that's supposed to be the order without the res, so if this test fail in the future people may think just needs to reorder the list and all good.

Reply:
It makes it readable, can't understand/see why that's supposed to be the order without the RHS, so if this test fail in the future people may think just needs to reorder the list and all good.